src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -357,6 +357,113 @@ public bool IsHomeSpace
             get { return Model == DynamoViewModel.HomeSpace; }
         }
 
+        /// <summary>
+        /// Returns the Uuid property of the seralized Workspace
+        /// </summary>
+        [JsonIgnore]
+        internal string GraphId { get; set; }
+
+        /// <summary>
+        /// Returns the Json representation of the current graph
+        /// </summary>
+        [JsonIgnore]
+        internal JObject JsonRepresentation { get; set; }
+
+        /// <summary>
+        /// Returns the stringified representation of the connected nodes
+        /// </summary>
+        [JsonIgnore]
+        public string Checksum
```

Comment:
1. This does not actually seem like a checksum - what does it look like when saved?
2. Have you evaluated taking the hash, md5, sha256 etc of either the entire graph, or just this string and serializing that instead of this entire thing? 
3. Related to above, it sounds like you actually want some collisions, which some of those hash functions will give you. You can hash the model or even just the serialized workspace. Take note though, that built-in `GetHashCode` of strings in .net6 will change from execution to execution - so they are not stable to serialize, use a proper known checksum function.

Reply:
> 1. This does not actually seem like a checksum - what does it look like when saved?
> 
>     2. Have you evaluated taking the hash, md5, sha256 etc of either the entire graph, or just this string and serializing that instead of this entire thing?
> 
>     3. Related to above, it sounds like you actually want some collisions, which some of those hash functions will give you. You can hash the model or even just the serialized workspace. Take note though, that built-in `GetHashCode` of strings in .net6 will change from execution to execution - so they are not stable to serialize, use a proper known checksum function.

1. The stringified value of its connected nodes, it's stored in the preferences. referring to https://jira.autodesk.com/browse/DYN-5879
2. We don't need the entire workspace, only it's anatomy (connected nodes) , that means a checksum. So if an user change the connections of the nodes from the same graph (substantial checksum) it should be send to Forge.
3. The building of the checksum is only grabbing the info from the connected nodes from a serialized workspace, their Ids don't change during an user session.